src/testing/h_refinement_study_isentropic_vortex.cpp

```diff
@@ -141,8 +172,12 @@ int HRefinementStudyIsentropicVortex<dim, nstate>::run_test() const
         pcout << "---------------------------------------------" << std::endl;
 
         const Parameters::AllParameters params = reinit_params_and_refine(refinement);
-        std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(&params, parameter_handler);
-        std::unique_ptr<FlowSolver::PeriodicEntropyTests<dim, nstate>> flow_solver_case = std::make_unique<FlowSolver::PeriodicEntropyTests<dim,nstate>>(&params);
+        auto params_modified = params;
+        if (nb_c_value > 0){
+            params_modified.FR_user_specified_correction_parameter_value = c_value;
```

Comment:
You could move the specification of the `c` parameter into the `reinit_params_and_refine` function. You would need to add extra arguments, `c_value` and `nb_c_value`.

Reply:
I like the renaming `params_modified`, much clearer than what I had initially chosen!